epinar/assignment1/q2.py

```diff
@@ -0,0 +1,65 @@
+class Node:
+
+    def __init__(self, data):
+        self.data = data
+        self.next = None
+
+
+class LinkedList:
+
+    def __init__(self, head):
+        self.size = 1
+        new_node = Node(head)
+        self.curr, self.head = new_node, new_node
+
+    def add(self, data):
+        new_node = Node(data)
+        if self.size == 0:
+            self.head = new_node
+        else:
+            self.curr.next = new_node
+
+        self.curr = new_node
+        self.size += 1
+
+        return True
+
+    def print(self):
```

Comment:
If you define the __repr__ method, then you can print a LinkedList without having to learn the method.

>>> ll = LinkedList()
>>> ll.add(1), ll.add(2)

Current solution:
>>> print ll.print()

If ll.print is renamed ll.__repr__
>>> print ll